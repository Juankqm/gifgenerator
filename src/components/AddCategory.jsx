import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) => {
        setInputValue(target.value)
    }
    const handleFormInfo = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <=1 ) return;
        onNewCategory(inputValue);
        setInputValue('');
    }
    return (
    <>
        <form onSubmit={handleFormInfo}>
        <input 
            type="text"
            placeholder="Search Gifs"
            value={inputValue}
            onChange={handleInputChange}
        />
        </form>
    </>
    )
}