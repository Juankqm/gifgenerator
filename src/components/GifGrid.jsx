import { useEffect ,useState} from 'react';
import {getGifs} from '../helpers/GetGifs';
const GifGrid =  ({category}) => {
    const [gifs, setGifs] = useState([{}]) 
    useEffect( () => { 
        getGifs(category)
    },[])
    return(
        <div>
            <h3>{category}</h3>   
        </div>
    );
}
export default GifGrid;