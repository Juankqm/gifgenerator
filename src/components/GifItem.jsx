export const GifItem = ({title,src}) => {
    return (
        <div className='card'>
            <img
                alt={title}
                src={src}
                loading="lazy" 
            />
            <p>{title}</p>
        </div>
    );
}
