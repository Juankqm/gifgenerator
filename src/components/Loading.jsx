export const Loading = () => {
    return(
    <>
        <h1>Loading Images..</h1>
    </>
    );
}
