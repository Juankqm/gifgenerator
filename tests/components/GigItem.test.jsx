import { render,screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem'

describe('Testing <GifItem />', () => {

    const title = 'Saitama';
    const url   = 'https://one-punch.com/saitama.jpg';

    test('should match snapshot', () => {
        const {container} = render(<GifItem title={title} src={url}/>);
        expect( container ).toMatchSnapshot();
    });

    test('Should show Image with URL & ALT indicated', () => {
        render(<GifItem title={title} src={url}/>);
        const { src,alt } = screen.getByRole('img');
        expect( src ).toBe( src );
        expect( alt ).toBe( alt );
    });

    test('should show Component title', () => {
        render(<GifItem title={title} src={url}/>);
        expect( screen.getByText( title )).toBeTruthy();
    })
});