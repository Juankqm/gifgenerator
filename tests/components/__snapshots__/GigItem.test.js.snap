// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      loading="lazy"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
